From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Taiyou06 <kaandindar21@gmail.com>
Date: Thu, 7 Nov 2024 21:50:47 +0100
Subject: [PATCH] Lithium: fast util

This patch is based on the following mixins:
* "net/caffeinemc/mods/lithium/mixin/math/fast_util/AABBMixin.java"
* "net/caffeinemc/mods/lithium/mixin/math/fast_util/AxisCycleDirectionMixin.java"
* "net/caffeinemc/mods/lithium/mixin/math/fast_util/DirectionMixin.java"
By: 2No2Name <2No2Name@web.de>
As part of: Lithium (https://github.com/CaffeineMC/lithium)
Licensed under: LGPL-3.0-only (https://www.gnu.org/licenses/lgpl-3.0.html)

// SYNC SOURCE START
rev: 01597fb879b7a6d17c2420ec4e25a98df3bc8e7b
branch: develop
repo: https://github.com/CaffeineMC/lithium.git
track:
- "common/src/main/java/net/caffeinemc/mods/lithium/mixin/math/fast_util/**"
// SYNC SOURCE END

diff --git a/net/minecraft/core/AxisCycle.java b/net/minecraft/core/AxisCycle.java
index e3524e706e02683d891a8de1e552a0bd0595b0f4..cac232cc56160239b4a82d9e2f91805fa405c007 100644
--- a/net/minecraft/core/AxisCycle.java
+++ b/net/minecraft/core/AxisCycle.java
@@ -35,7 +35,17 @@ public enum AxisCycle {
 
         @Override
         public Direction.Axis cycle(final Direction.Axis axis) {
-            return AXIS_VALUES[Math.floorMod(axis.ordinal() + 1, 3)];
+            // Leaf start - Lithium - fast util
+            return switch (axis.ordinal()) {
+                case 0 -> //X
+                    Direction.Axis.Y;
+                case 1 -> //Y
+                    Direction.Axis.Z;
+                case 2 -> //Z
+                    Direction.Axis.X;
+                default -> throw new IllegalArgumentException();
+            };
+            // Leaf end - Lithium - fast util
         }
 
         @Override
@@ -56,7 +66,17 @@ public enum AxisCycle {
 
         @Override
         public Direction.Axis cycle(final Direction.Axis axis) {
-            return AXIS_VALUES[Math.floorMod(axis.ordinal() - 1, 3)];
+            // Leaf start - Lithium - fast util
+            return switch (axis.ordinal()) {
+                case 0 -> //X
+                    Direction.Axis.Z;
+                case 1 -> //Y
+                    Direction.Axis.X;
+                case 2 -> //Z
+                    Direction.Axis.Y;
+                default -> throw new IllegalArgumentException();
+            };
+            // Leaf end - Lithium - fast util
         }
 
         @Override
diff --git a/net/minecraft/core/Direction.java b/net/minecraft/core/Direction.java
index 46861ad98049f363d59e17367f3fc1f948ab15b5..e6b14aa7f9bd3003428c4e5756b9daf6a14d1a49 100644
--- a/net/minecraft/core/Direction.java
+++ b/net/minecraft/core/Direction.java
@@ -363,7 +363,7 @@ public enum Direction implements Directional, StringRepresentable, ca.spottedlea
     }
 
     public static Direction getRandom(final RandomSource random) {
-        return Util.getRandom(VALUES, random);
+        return VALUES[random.nextInt(VALUES.length)]; // Leaf - Lithium - fast util
     }
 
     public static Direction getApproximateNearest(final double dx, final double dy, final double dz) {
diff --git a/net/minecraft/world/phys/AABB.java b/net/minecraft/world/phys/AABB.java
index 36780d280c96f44be0e7d0dcd3467fff5e843824..350ec6796abc4d53627871b35651a2490e4ac175 100644
--- a/net/minecraft/world/phys/AABB.java
+++ b/net/minecraft/world/phys/AABB.java
@@ -80,11 +80,32 @@ public class AABB {
     }
 
     public double min(final Direction.Axis axis) {
-        return axis.choose(this.minX, this.minY, this.minZ);
+        // Leaf start - Lithium - fast util
+        return switch (axis.ordinal()) {
+            case 0 -> //X
+                this.minX;
+            case 1 -> //Y
+                this.minY;
+            case 2 -> //Z
+                this.minZ;
+            default -> throw new IllegalArgumentException();
+        };
+
+        // Leaf end - Lithium - fast util
     }
 
     public double max(final Direction.Axis axis) {
-        return axis.choose(this.maxX, this.maxY, this.maxZ);
+        // Leaf start - Lithium - fast util
+        return switch (axis.ordinal()) {
+            case 0 -> //X
+                this.maxX;
+            case 1 -> //Y
+                this.maxY;
+            case 2 -> //Z
+                this.maxZ;
+            default -> throw new IllegalArgumentException();
+        };
+        // Leaf end - Lithium - fast util
     }
 
     @Override
